spec/client/ics-009-loopback-cilent/README.md

```diff
@@ -0,0 +1,244 @@
+---
+ics: 9
+title: Loopback Client
+stage: draft
+category: IBC/TAO
+kind: instantiation
+author: Christopher Goes <cwgoes@tendermint.com>
+created: 2020-01-17
+modified: 2020-01-17
+requires: 2
+implements: 2
+---
+
+## Synopsis
+
+This specification describes a loopback client, designed to be used for interaction over the IBC interface with modules present on the same ledger.
+
+### Motivation
+
+Loopback clients may be useful in cases where the calling module does not have prior knowledge of where precisely the destination module lives and would like to use the uniform IBC message-passing interface (similar to `127.0.0.1` in TCP/IP).
+
+### Definitions
+
+Functions & terms are as defined in [ICS 2](../../core/ics-002-client-semantics).
+
+`ConnectionEnd` and `generateIdentifier` are as defined in [ICS 3](../../core/ics-003-connection-semantics)
+
+`getCommitmentPrefix` and `removePrefix` are as defined in [ICS 24](../../core/ics-024-host-requirements).
+
+### Desired Properties
+
+Intended client semantics should be preserved, and loopback abstractions should be negligible cost.
+
+## Technical Specification
+
+### Data structures
+
+No consensus state, headers, or evidence data structures are required for a loopback client. The loopback client does not need to store the consensus state of a remote chain, since state verification does not require to check a Merkle proof against a previously validated commitment root.
+
+```typescript
+type ConsensusState object
+
+type Header object
+
+type Misbehaviour object
+```
+
+### Client state
+
+The loopback client state tracks the latest height of the local ledger.
+
+```typescript
+interface ClientState {
+  latestHeight: Height
+}
+```
+
+### Height
+
+The height of a loopback client state consists of two `uint64`s: the revision number, and the height in the revision.
+
+```typescript
+interface Height {
+  revisionNumber: uint64
+  revisionHeight: uint64
+}
+```
+
+### Header
+
+The header includes the identifier of the light client and the current block height of the local ledger.
+
+```typescript
+interface Header {
```

Comment:
I know that we don't have a `Header` type in ibc-go for the localhost client, but I thought that it was nice as an abstraction, so that we don't need to specify a `GetSelfHeight` function that is used in `UpdateState`. Happy to change this though, if folks think otherwise.